tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -3309,38 +3309,26 @@ LogicalResult ConvertTFLSinOp::matchAndRewrite(
   ShapedType output_type =
       dyn_cast<ShapedType>(tfl_sin_op.getResult().getType());
 
-  std::optional<Value> result = convertSinOp(rewriter, op, input, output_type);
-  if (!result) return failure();
+  if (!output_type)
+    return rewriter.notifyMatchFailure(op, "output_type required");
+  CreateReplaceOpAndInfer<tosa::SinOp>(rewriter, op, output_type,
+                                       tfl_sin_op.getX());
 
-  rewriter.replaceOp(op, {result.value()});
   return success();
 }
 
 LogicalResult ConvertTFLCosOp::matchAndRewrite(
     Operation* op, PatternRewriter& rewriter) const {
   auto tfl_cos_op = cast<TFL::CosOp>(op);
   Value input = tfl_cos_op.getX();
```

Comment:
 Missed it (but presubmit caught) that this one is unused.

Reply:
thanks! removed this in the latest patch set.